### PR TITLE
Ensure "WasRedirected" Calculation is Correct

### PR DESCRIPTION
**Pull request by mborof**
This makes the PayClient self reliant in determining if the user is returning from a popup or a redirect and removes code from the analytics client that was being used to make this determination.

### CODE REVIEW

File: src/runtime/analytics-service-test.js [hunk 1]
```diff
@@ -118,6 +118,13 @@ describes.realWin('AnalyticsService', {}, env => {
       analyticsService.setTransactionId(txId);
       expect(analyticsService.getTransactionId()).to.equal(txId);
     });
+
+    it('should close', () => {
+      const activityIframe = analyticsService.getElement();
+      expect(activityIframe.parentNode === runtime.doc().getBody()).to.true;
```
Comment (henel677) [inline src/runtime/analytics-service-test.js]: Is it possible to use:
expect(...).to.equal(...) instead
?

File: src/runtime/analytics-service-test.js [hunk 2]
```diff
@@ -287,6 +294,26 @@ describes.realWin('AnalyticsService', {}, env => {
...
+
+    it('Should work without waiting for the promise', async () => {
+      // This sends another event and waits for it to be sent
+      eventManagerCallback({
+        eventType: AnalyticsEvent.IMPRESSION_PAYWALL,
+        eventOriginator: EventOriginator.SWG_CLIENT,
+        isFromUserAction: true,
+        additionalParameters: {droppedData: true},
+      });
+      await analyticsService.lastAction_;
```
Comment (henel677) [inline src/runtime/analytics-service-test.js]: Does it make sense to verify that that lastAction_ is not null before waiting for it (apparently waiting for non-promises is valid and returns the non-promise)?

File: src/runtime/pay-flow.js [hunk 3]
```diff
@@ -143,6 +143,7 @@ export class PayStartFlow {
       true,
       getEventParams(swgPaymentRequest['skuId'])
     );
+    PayCompleteFlow.waitingForGpay_ = true;
```
Comment (alin04) [inline src/runtime/pay-flow.js]: To be pedantic, this variable is being set a bit earlier than actual gpay invocation. But I suppose it's inconsequential.

Comment (mborof) [reply src/runtime/pay-flow.js]: Should I change the flag to "waitingForPayClient_"?

Comment (alin04) [reply src/runtime/pay-flow.js]: sure

File: src/runtime/analytics-service-test.js [hunk 4]
```diff
@@ -119,6 +119,13 @@ describes.realWin('AnalyticsService', {}, env => {
       analyticsService.setTransactionId(txId);
       expect(analyticsService.getTransactionId()).to.equal(txId);
     });
+
+    it('should close', () => {
+      const activityIframe = analyticsService.getElement();
+      expect(activityIframe.parentNode).to.equal(runtime.doc().getBody());
+      analyticsService.close();
+      expect(activityIframe.parentNode === null).to.true;
```
Comment (henel677) [inline src/runtime/analytics-service-test.js]: I think this could also use "....to.not.be.null;"

**Pull request merged**